notify/notify.go

```diff
@@ -225,11 +226,16 @@ func newMetrics(r prometheus.Registerer) *metrics {
 			Name:      "notifications_total",
 			Help:      "The total number of attempted notifications.",
 		}, []string{"integration"}),
-		numFailedNotifications: prometheus.NewCounterVec(prometheus.CounterOpts{
+		numTotalFailedNotifications: prometheus.NewCounterVec(prometheus.CounterOpts{
 			Namespace: "alertmanager",
 			Name:      "notifications_failed_total",
 			Help:      "The total number of failed notifications.",
 		}, []string{"integration"}),
+		numPermanentlyFailedNotifications: prometheus.NewCounterVec(prometheus.CounterOpts{
+			Namespace: "alertmanager",
+			Name:      "notifications_permanently_failed_total",
```

Comment:
Not totally convinced by the metric name. We should also count the number of tries.
I'm wondering if we shouldn't use the existing `alertmanager_notifications_failed_total` to count notifications that couldn't be delivered permanently and introduce `alertmanager_notification_requests_total`/`alertmanager_notification_requests_failed_total` metrics to count notification request attempts and failures.

Reply:
@simonpasquier thanks for the update. I tried to be non-intrusive and keep the existing meaning for the `notifications_failed_total`. I agree that your suggested metrics have much clearer intent given these names. 
Should I introduce them and delete the old metric or just add the new ones?